src/device-state/preload.ts

```diff
@@ -154,7 +156,7 @@ export async function loadTargetFromFile(appsPath: string): Promise<boolean> {
 		}
 	} finally {
 		const targetPath = appsJsonBackup(appsPath);
-		if (!(await fsUtils.exists(targetPath))) {
+		if (isTargetLoaded && !(await fsUtils.exists(targetPath))) {
```

Comment:
I don't think it makes so much sense to do this check within the `finally`. I think we can probably move the whole `await fsUtils.safeRename...` call to after the `setTarget` call and we can avoid this variable. 

Since the call is a bit verbose, we can put it in an function and call that after the `setTarget`.

Reply:
Good idea, will change

Reply:
Changed as of https://github.com/balena-os/balena-supervisor/pull/1944/commits/d92d696e0e599fe367729df679031a37d9503401